openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java

```diff
@@ -131,6 +142,31 @@ private User checkAndCreateUser(String email) throws IOException {
     }
   }
 
+  /**
+   * Check if the user exists in database by userName, if user exist, reassign roles for user according to it's ldap
+   * group else, create a new user and assign roles according to it's ldap group
+   *
+   * @param email email address of user
+   * @param userName userName of user
+   * @param userDn the dn of user from ldap
+   * @return user info
+   * @author Eric Wen@2023-07-16 17:06:43
+   */
+  private User checkAndCreateUser(String email, String userName, String userDn) throws IOException {
```

Comment:
I think this could be used as common method since we are using it for Admin creation as well cc @mohityadav766